### DISCUSSION TITLE
KKS multi phases

### DISCUSSION THREAD

**Discussion by LiuPengPeter** [idaholab/moose#16495] [Q&A Modules: Phase field] [answered]
Hello everyone,

I have some questions about the kernel of multi phase. Could anyone help?
 
1) For KKS module multi phases model, there is the constraint for order parameters. eta1+eta2+eta3_eta4=1. So n-1 AC equations are needed. WHile, for WBM, every order parameter needs a AC equation.

2) For  both KKS and WBM multi phases model, the constraint h1+h2+h3+h4=1 must be added by lagrange multiplier technique or free energy penalty term.  If I use the KKS model， which method is better? SwitchingFunctionPenalty (energy penalty term) seems easier.

3) Could the split form of CH be used in KKS multi phase? 

Thanks.

**Comment by souravmat-git** [comment, marked as answer]
Hello,

There are two kinds of "KKS type" multi-phase-field model—the first is a model by Eiken et al. (see Ref. https://journals.aps.org/pre/abstract/10.1103/PhysRevE.73.066122) and the other is a model by Moelans (see Ref. https://www.sciencedirect.com/science/article/pii/S1359645410007019).

Answer to your second question:

Eiken's model will require the constraint on interpolation functions h1+h2+h3+h4=1. However, Moelan's model does not need any such constraint due to the definition of interpolation function. This interpolation function is available in moose (check the material moose/modules/phase_field/src/materials/SwitchingFunctionMultiPhaseMaterial.C)

You can also use this function for multi-phase-field "WBM type" model.

Answer to your first question:

Contrary to Eiken's model, the constraint on eta is not needed in Moelan's model. In Eiken's model this constraint is due to the fact that the interpolation function is equal to the phase-field variable, i.e., h1 = eta1.

Answer to your last question:

To my understanding, in KKS we simply solve for Fick's diffusion equation (see the Refs.), and therefore the split form is not needed.
Moreover, this equation is already implemented as a moose kernel (check moose/framework/src/kernels/MatDiffusion.C). 
Also, check the example input file moose/modules/phase_field/test/tests/KKS_system/kks_multiphase.i

Hope it helps.
Sourav
